### PR TITLE
fix(elixir): Refactor and improvement of architecture

### LINKED ISSUE #47: enhancement(elixir): Improve GenServer architecture
**Type**

Enhancement

**Current behaviour**
Right now all of the game rooms are supervised by one GenServer

**Expected behaviour**
We could try to change the architecture to the one suggested in the following article:
[multiple gen servers](http://codeloveandboards.com/blog/2016/03/20/supervising-multiple-genserver-processes/)

**Motivation / use case**
  * If one process dies, in the current situation we will lose data for all of the rooms. In the enhanced version only one of the rooms will have to be reset. 
  * Performance issues: moving from one process having to perform operations on all of the rooms to many processes each having to take care of only one game room

### PR DESCRIPTION
**Pull request by mrapacz**
**Summary**
This PR is a response to #47 with an attempt to improve current codebase's architecture.

**Related issues**
closes: #47 
does half of the job for: #52 

**Test plan**

### CODE REVIEW

**Comment by mrapacz** [comment]
Decided not to include type annotations in this PR, can do it possibly in a separate one

File: aion/lib/aion.ex [hunk 1]
```diff
@@ -16,9 +14,8 @@ defmodule Aion do
       supervisor(Repo, []),
       # Start the endpoint when the application starts
       supervisor(Endpoint, []),
-      # Start your own worker by calling: Aion.Worker.start_link(arg1, arg2, arg3)
-      # worker(Aion.Worker, [arg1, arg2, arg3]),
-      worker(ChannelMonitor, [%{}])
+
```
Comment (pmrukot) [inline aion/lib/aion.ex]: This newline seems out of place. I would also delete the comments.

File: aion/test/lib/room_channel/user_record_test.exs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  alias Aion.RoomChannel.UserRecord
+
+  @user_record %UserRecord{username: "Jack"}
+
+  test "update score" do
+    assert @user_record.score == 0
+    assert UserRecord.update_score(@user_record) == %UserRecord{@user_record | score: 1}
+    assert UserRecord.update_score(@user_record, 10) == %UserRecord{@user_record | score: 10}
+  end
+end
```
Comment (pmrukot) [inline aion/test/lib/room_channel/user_record_test.exs]: Missing newline.

File: aion/test/models/question_test.exs [hunk 3]
```diff
@@ -15,4 +15,17 @@ defmodule Aion.QuestionTest do
...
+  def get_question(question_id) do
+    Repo.get(Question, question_id)
+  end
+
+  def get_random_question(category_id) do
+    query = from q in Question, where: q.subject_id == ^category_id
+    question =
+      query
+      |> Repo.all()
+      |> Enum.random()
```
Comment (pmrukot) [inline aion/test/models/question_test.exs:28]: Here you omit the parentheses: https://github.com/pmrukot/aion/pull/67/files#diff-fcb2b290b530eae3a0bbb6b044f1cfbeR32

I think we need to settle down on way of piping through one-arity functions. Elixir style guide recommends doing so but Im not fully convinced:
https://github.com/christopheradams/elixir_style_guide#parentheses-pipe-operator

**Review by pmrukot — APPROVED** [review]
Absolutely fantastic job 👏 👏 👏

File: aion/test/models/question_test.exs (hunk 3 above)
Comment (mrapacz) [reply aion/test/models/question_test.exs:28]: I think we can keep with the style and not omit them, looks good imho.

**Pull request merged**